### PR TITLE
arrow-cpp: disable some potentially failing s3fs tests

### PR DESCRIPTION
**Pull request by cpcloud**
###### Motivation for this change

These tests can hang or fail, disable them when running arrow-cpp tests.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/arrow-cpp/default.nix [hunk 1]
```diff
@@ -171,11 +171,39 @@ stdenv.mkDerivation rec {
...
+        "TestS3FSGeneric.GetFileInfoVector"
+        "TestS3FSGeneric.MoveFile"
+        "TestS3FSGeneric.OpenInputFile"
+        "TestS3FSGeneric.OpenInputFileAsync"
+        "TestS3FSGeneric.OpenInputFileWithFileInfo"
+        "TestS3FSGeneric.OpenInputStream"
+        "TestS3FSGeneric.OpenInputStreamAsync"
+        "TestS3FSGeneric.OpenInputStreamWithFileInfo"
+        "TestS3FSGeneric.OpenOutputStream"
+        "TestS3FSGeneric.SpecialChars"
```
Comment (veprbl) [inline pkgs/development/libraries/arrow-cpp/default.nix:206]: I believe, gtest supports globbing.
```suggestion
        "TestS3FSGeneric.*"
```

**Comment by veprbl** [comment]
Also, at this point we can drop `minio` dependency?

**Comment by cpcloud** [comment]
@veprbl Unfortunately we can't remove the `minio` dependency. Without it, `arrow-s3fs-test` will fail to start at all.

**Comment by veprbl** [comment]
> @veprbl Unfortunately we can't remove the `minio` dependency. Without it, `arrow-s3fs-test` will fail to start at all.

Thank you for checking!

**Pull request merged**